app/pulp/app/serializers/repository.py

```diff
@@ -37,3 +38,34 @@ class Meta:
         fields = ModelSerializer.Meta.fields + ('name', 'description', 'notes',
                                                 'last_content_added', 'last_content_removed')
 
+
+class ImporterSerializer(MasterModelSerializer):
+    """
+    Base class for Importers, this is intended to be subclassed by each plugin.
+
+    Please import from `pulp.app.serializers` rather than from this module directly.
+
+    Every class that inherrits from this class must override the `_href` method with a
+    `RepositoryNestedIdentityField` that defines the view_name.
+
+    """
+    name = serializers.CharField(
+        help_text='A name for this importer, unique within the associated repository.'
+    )
+
+    class Meta:
+        abstract = True
+        fields = MasterModelSerializer.Meta.fields + ('name',)
+
+
+class ExampleDetailImporterSerializer(ImporterSerializer):
+    """
+    Example of a Detail serializer that also utilizes nested routes.
+    """
+    # Content adapters need to use two kwargs because the name is only unique within a
+    # repository and alone, `Importer.name` is not enough to uniquely identify it.
```

Comment:
This comment doesn't make sense to me in this context. It talks about needing to use two kwargs, but only one kwarg is used on the next line. Probably reword this to be more like "Serializers for ContentAdapter models have a special identity that is only valid in the context of the repositories that they're related to, which RepositoryNestedIdentityField knows how to handle", or something.